app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt

```diff
@@ -305,11 +305,11 @@ class ValueStoreImpl(
 
     private fun deleteAttributeValue(field: String, optionUid: String): StoreResult {
         val option = d2.optionModule().options().uid(optionUid).blockingGet()
-        val possibleValues = arrayListOf(option.name()!!, option.code()!!)
+        val possibleValues = arrayListOf(option?.name(), option?.code()).filterNotNull()
```

Comment:
I think that this methods always returns false is option is null, so maybe you can return false after the line 307 if it is null.